Source/FakeItEasy/Priority.cs

```diff
@@ -0,0 +1,193 @@
+    /// Indicates precedence between otherwise indistinguishable options, for example when deciding
+    /// which <see cref="IDummyFactory"/> should be used to create a Dummy of a given type.
+    /// In making such decisions, the higher-valued <c>Priority</c> will be used.
+    /// </summary>
+    public struct Priority : IComparable<Priority>, IEquatable<Priority>
+    {
+        private static readonly Priority DefaultInstance = new Priority(0);
+
+        private static readonly Priority InternalInstance = new Priority(-1);
+
+        private readonly int value;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Priority"/> struct.
+        /// </summary>
+        /// <param name="value">The value of the <c>Priority</c>.</param>
+        /// <returns>A new <c>Priority</c> with the specified value.</returns>
+        public Priority(byte value)
+        {
+            this.value = value;
+        }
+
+        private Priority(int value)
+        {
+            this.value = value;
+        }
+
+        /// <summary>
+        /// Gets the default <c>Priority</c> used by client-supplied <see cref="IDummyFactory"/>s,
+        /// <see cref="IFakeOptionsBuilder"/>s, and <see cref="IArgumentValueFormatter"/>s.
+        /// </summary>
+        /// <remarks>Equivalent to a Priority with value <c>0</c>.</remarks>
+        public static Priority Default
+        {
+            get { return DefaultInstance; }
+        }
+
+        /// <summary>
+        /// Gets the <c>Priority</c> used by all FakeItEasy-supplied <see cref="IDummyFactory"/>s,
+        /// <see cref="IFakeOptionsBuilder"/>s, and <see cref="IArgumentValueFormatter"/>s.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Equivalent to a Priority with value <c>-1</c>.
+        /// </para>
+        /// <para>
+        /// We have a single such Priority because it's all that's currently needed, and
+        /// we're unlikely to introduce built-in dummy factories, fake options builders, or
+        /// argument value formatters that would collide with one another.
+        /// Just in case, though, all negative values have been reserved for use by FakeItEasy.
+        /// </para>
+        /// </remarks>
+        internal static Priority Internal
+        {
+            get { return InternalInstance; }
+        }
+
+        /// <summary>
+        /// Compares <see paramref="left"/> and <see paramref="right"/>, returning <c>true</c> if
+        /// and only if the value of <c>left</c> is less than that of <c>right</c>.
+        /// </summary>
+        /// <param name="left">One priority to compare.</param>
+        /// <param name="right">The other priority to compare.</param>
+        /// <returns><c>true</c> if and only if the value of <c>left</c> is less than that of <c>right</c>.</returns>
+        public static bool operator <(Priority left, Priority right)
+        {
+            return left.CompareTo(right) < 0;

```

Comment:
Would you like me to do the same in `Equals(object)`? Otherwise, I can continue to delegate to `Equals(Priority)`.